src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/editors/media.controller.js

```diff
@@ -1,112 +1,129 @@
 angular.module("umbraco")
     .controller("Umbraco.PropertyEditors.Grid.MediaController",
-    function ($scope, $timeout, userService, editorService) {
-        
-        
-        $scope.thumbnailUrl = getThumbnailUrl();
+    function ($scope, userService, editorService, localizationService) {        
         
+        $scope.thumbnailUrl = getThumbnailUrl();        
         
         if (!$scope.model.config.startNodeId) {
             if ($scope.model.config.ignoreUserStartNodes === true) {
                 $scope.model.config.startNodeId = -1;
                 $scope.model.config.startNodeIsVirtual = true;
 
             } else {
-                userService.getCurrentUser().then(function (userData) {
+                userService.getCurrentUser().then(userData => {
                     $scope.model.config.startNodeId = userData.startMediaIds.length !== 1 ? -1 : userData.startMediaIds[0];
                     $scope.model.config.startNodeIsVirtual = userData.startMediaIds.length !== 1;
                 });
             }
         }
-        
-        $scope.setImage = function(){
+
+        $scope.setImage = function() {
             var startNodeId = $scope.model.config && $scope.model.config.startNodeId ? $scope.model.config.startNodeId : undefined;
-            var startNodeIsVirtual = startNodeId ? $scope.model.config.startNodeIsVirtual : undefined;
-            var value = $scope.control.value;
-            var target = value
-                ? {
-                    udi: value.udi,
-                    url: value.image,
-                    image: value.image,
-                    focalPoint: value.focalPoint,
-                    coordinates: value.coordinates
-                }
-                : null;
+
             var mediaPicker = {
                 startNodeId: startNodeId,
-                startNodeIsVirtual: startNodeIsVirtual,
+                startNodeIsVirtual: startNodeId ? $scope.model.config.startNodeIsVirtual : undefined,
```

Comment:
@nathanwoulfe I think it should be `null` instead of `undefined` here since you are assigning a value to the property.

Reply:
yup, it should be. That was a bit of refactoring to get rid of an unnecessary variable, didn't check that the statement made sense. Will update.

Reply:
Cool, I guess it is the same case with `startNodeId` variable.

Reply:
I've updated to use null 👍